spec/services/search/search_service_spec.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Search::SearchService, type: :service do
+  describe '#initialize' do
+    context 'given the search engine is google' do
+      it 'returns instance of GoogleSearchService' do
+        search_service = described_class.new(search_engine: 'google', keyword: 'ruby')
+        expect(search_service).to be_a(Search::GoogleSearchService)
+      end
+    end
+
+    context 'given the search engine is INVALID' do
+      it 'raises an error' do
+        expect { described_class.new(search_engine: 'yahoo', keyword: 'ruby') }.to raise_error(NotImplementedError)
+      end
+    end
+  end
+
+  describe '#search' do
```

Comment:
Shouldn't this test be defined in another test file for `GoogleSearchService`? That instance method does not exist on `GoogleSearchService`.